packages/packages/core/editor-canvas/src/legacy/create-inline-editing-element-type.tsx

```diff
@@ -0,0 +1,198 @@
+import * as React from 'react';
+import { createRoot, type Root } from 'react-dom/client';
+import { InlineEditor } from '@elementor/editor-controls';
+import { getElementType } from '@elementor/editor-elements';
+import {
+	htmlPropTypeUtil,
+	stringPropTypeUtil,
+	type StringPropValue,
+	type TransformablePropValue,
+} from '@elementor/editor-props';
+import { ThemeProvider } from '@elementor/editor-ui';
+
+import {
+	getBlockedValue,
+	getHtmlPropType,
+	getInitialPopoverPosition,
+	getInlineEditablePropertyName,
+	getWidgetType,
+	legacyWindow,
+} from '../utils/inline-editing-utils';
+import { type CreateTemplatedElementTypeOptions, createTemplatedElementView } from './create-templated-element-type';
+import { type ElementType, type ElementView } from './types';
+
+export function createInlineEditingElementType( {
+	type,
+	renderer,
+	element,
+}: CreateTemplatedElementTypeOptions ): typeof ElementType {
+	return class extends legacyWindow.elementor.modules.elements.types.Widget {
+		getType() {
+			return type;
+		}
+
+		getView() {
+			return createInlineEditingElementView( {
+				type,
+				renderer,
+				element,
+			} );
+		}
+	};
+}
+
+export function createInlineEditingElementView( {
+	type,
+	renderer,
+	element,
+}: CreateTemplatedElementTypeOptions ): typeof ElementView {
+	const TemplatedView = createTemplatedElementView( { type, renderer, element } );
+
+	Object.entries( element.twig_templates ).forEach( ( [ key, template ] ) => {
+		renderer.register( key, template );
+	} );
+
+	return class extends TemplatedView {
+		inlineEditorRoot: Root | null = null;
+		handlerAttached = false;
+
+		render() {
+			if ( this.inlineEditorRoot ) {
+				this.resetInlineEditorRoot();
+			}
+
+			if ( ! this.isValueDynamic() && ! this.handlerAttached ) {
+				this.$el.on( 'dblclick', '*', this.handleRenderInlineEditor.bind( this ) );
```

Comment:
Nitpick: Are we sure we want to use jQuery here? This is inside a react component? Can we not get a reference to an element the 'react' way?